### PR TITLE
Update docs to generalize from `DANDI Archive` to `DANDI instance`

### PR DESCRIPTION
**Pull request by kabilar**
- [x] Builds on #1527 to generalize the docs to describe that the client allows for interacting with `DANDI instances, such as the DANDI Archive`
- [x] Update `Dandi` to `DANDI`
- [x] Update `DANDI archive` to `DANDI Archive`
- [x] Update `DANDI Archive instances` to `DANDI instances`

cc @aaronkanzer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ Commands:
   delete            Delete dandisets and assets from the server.
   digest            Calculate file digests
   download          Download a file or entire folder from DANDI.
-  instances         List known Dandi Archive instances that the CLI can...
+  instances         List known DANDI instances that the CLI can...
```
Comment (yarikoptic) [inline README.md:68]:
```suggestion
  instances         List known DANDI instances that the CLI can interact
```

File: dandi/cli/tests/data/update_dandiset_from_doi/biorxiv.json [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
+    "citation": "Aguillon-Rodriguez, Valeria; Angelaki, Dora E.; Bayer, Hannah M.; Bonacchi, Niccol\u00f2; Carandini, Matteo; Cazettes, Fanny; Chapuis, Gaelle A.; Churchland, Anne K.; Dan, Yang; Dewitt, Eric E. J.; Faulkner, Mayo; Forrest, Hamish; Haetzel, Laura M.; Hausser, Michael; Hofer, Sonja B.; Hu, Fei; Khanal, Anup; Krasniak, Christopher S.; Laranjeira, In\u00eas; Mainen, Zachary F.; Meijer, Guido T.; Miska, Nathaniel J.; Mrsic-Flogel, Thomas D.; Murakami, Masayoshi; Noel, Jean-Paul; Pan-Vazquez, Alejandro; Rossant, Cyrille; Sanders, Joshua I.; Socha, Karolina Z.; Terry, Rebecca; Urai, Anne E.; Vergara, Hernando M.; Wells, Miles J.; Wilson, Christian J.; Witten, Ilana B.; Wool, Lauren E.; Zador, Anthony (2023) Standardized and reproducible measurement of decision-making in mice (Version draft) [Data set]. DANDI Archive. http://localhost:8085/dandiset/000001/draft",
```
Comment (yarikoptic) [inline dandi/cli/tests/data/update_dandiset_from_doi/biorxiv.json:16]: those are coming from dandi-archive

```
❯ git grep 'DANDI archive'
dandiapi/api/models/version.py:        citation = f'{name} ({year}). (Version {version}) [Data set]. DANDI archive. {url}'
dandiapi/api/models/version.py:                    f'{cl} ({year}) {name} (Version {version}) [Data set]. DANDI archive. {url}'
...
```

there should be a "satellite" PR there to likely preceed these changes or we should improve testing here to allow for flexibility (didn't look ATM) but these are the cause of failing tests ATM e.g. https://github.com/dandi/dandi-cli/actions/runs/12262974763/job/34213512790?pr=1548#step:8:25905

File: dandi/dandiapi.py [hunk 3]
```diff
@@ -397,7 +397,7 @@ def get_page(pageno: int) -> list:
 
 
 class DandiAPIClient(RESTFullAPIClient):
-    """A client for interacting with a Dandi Archive server"""
+    """A client for interacting with a DANDI server"""
```
Comment (yarikoptic) [inline dandi/dandiapi.py:400]: Just for completeness
```suggestion
    """A client for interacting with a DANDI API server"""
```

**Review by yarikoptic — CHANGES_REQUESTED** [review]
This is great!  See some minor comments, but major thing I think is to look either to make testing there more flexible (allow for regex matching?) or just remove those changes from this PR and coordinate later with similar changes to dandi-archive

File: dandi/cli/tests/data/update_dandiset_from_doi/biorxiv.json (hunk 2 above)
Comment (yarikoptic) [reply dandi/cli/tests/data/update_dandiset_from_doi/biorxiv.json:16]: do you think you could remove those changes in the tests for now so we could not delay accepting major portion of the changes @kabilar to not breed conflicts etc?

Comment (kabilar) [reply dandi/cli/tests/data/update_dandiset_from_doi/biorxiv.json:16]: Hi @yarikoptic, I just created https://github.com/dandi/dandi-archive/pull/2114.

Comment (yarikoptic) [reply dandi/cli/tests/data/update_dandiset_from_doi/biorxiv.json:16]: it was merged now, so we should be all set here.  I will retrigger tests to see if all good (frankly I don't remember if we need release of archive first) and then we could proceed

**Comment by yarikoptic** [comment]
oh that codecov

codecov/patch — 0.00% of diff hit (target 88.44%) 

;-)

**Pull request merged**